src/model/CoilWaterHeatingDesuperheater.cpp

```diff
@@ -413,20 +377,64 @@ namespace model {
       OS_ASSERT(result);
     }
 
-    // bool CoilWaterHeatingDesuperheater_Impl::setTank(const HVACComponent& waterHeater) {
-    //   bool result = setPointer(OS_Coil_WaterHeating_DesuperheaterFields::TankName, waterHeater.handle());
-    //   if (result) addToWaterHeater(waterHeater);
-    //   return result;
-    // }
+    bool CoilWaterHeatingDesuperheater_Impl::setRatedHeatReclaimRecoveryEfficiency(double ratedHeatReclaimRecoveryEfficiency) {
+
+      if (auto heatingSource_ = heatingSource()) {
+
+        auto heatingSourceIddObjectType = heatingSource_->iddObjectType();
+        if ((heatingSourceIddObjectType == IddObjectType::OS_Refrigeration_Condenser_AirCooled)
+            || (heatingSourceIddObjectType == IddObjectType::OS_Refrigeration_Condenser_EvaporativeCooled)
+            || (heatingSourceIddObjectType == IddObjectType::OS_Refrigeration_Condenser_WaterCooled)) {
+          if (ratedHeatReclaimRecoveryEfficiency > 0.9) {
+            LOG(Error, briefDescription() << ": Rated Heat Reclaim Recovery Efficiency must be"
+                                          << " <= 0.9 when Heating Source Object Type = " << heatingSourceIddObjectType.valueDescription() << ".");
+            return false;
+          }
+        } else {
+          if (ratedHeatReclaimRecoveryEfficiency > 0.3) {
+            LOG(Error, briefDescription() << ": Rated Heat Reclaim Recovery Efficiency must be"
+                                          << " <= 0.3 when Heating Source Object Type = " << heatingSourceIddObjectType.valueDescription() << ".");
+            return false;
+          }
+        }
+      }
+
+      bool result = setDouble(OS_Coil_WaterHeating_DesuperheaterFields::RatedHeatReclaimRecoveryEfficiency, ratedHeatReclaimRecoveryEfficiency);
+      return result;
+    }
```

Comment:
Enforce varying limit on setRatedHeatReclaimRecoveryEfficiency if there is already a Heating Source assigned